### PR TITLE
docs: add example for awscc_eks_cluster

### PR DESCRIPTION
**Pull request by maheshr-amzn**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

### Description
First attempt to improve documentation for resource awscc_eks_cluster. I reused the existing tfplugindocs structure for both templates and examples.

Relates OR Closes #0000

### CODE REVIEW

File: docs/resources/eks_cluster.md [hunk 1]
```diff
@@ -10,7 +9,393 @@ description: |-
...
+  description = "EKS Cluster Version"
+  type        = string
+  default     = "1.27"
+}
+
+variable "eks_cluster_subnets" {
+  description = "Subnets for EKS Cluster"
+  type        = list(string)
+  default     = ["subnet-xxxx", "subnet-yyyy"] // Provide a list of subnet-ids for Amazon EKS Cluster
+}
```
Comment (jackywong-amazon) [inline docs/resources/eks_cluster.md:53]: Can we remove the variable and only show the core configuration?

File: docs/resources/eks_cluster.md [hunk 2]
```diff
@@ -10,7 +9,393 @@ description: |-
...
+
+resource "awscc_iam_role" "main" {
+  description = "IAM Role of EKS Cluster"
+  role_name   = "${var.eks_cluster_name}-role"
```
Comment (jackywong-amazon) [inline docs/resources/eks_cluster.md:57]: role_name = "example-role"

File: docs/resources/eks_cluster.md [hunk 3]
```diff
@@ -10,7 +9,393 @@ description: |-
...
+  tags = concat([ 
+    {
+       key   = "Name"
+       value = "${var.eks_cluster_name}-role"
+    } ],
+    var.eks_default_tags)
+}
+
+resource "awscc_eks_cluster" "main" {
+  name     = var.eks_cluster_name
```
Comment (jackywong-amazon) [inline docs/resources/eks_cluster.md:85]: name = "example_cluster_name"

File: examples/resources/awscc_eks_cluster/eks_cluster_basic.tf [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  type        = string
+  default     = "1.27"
+}
+
+variable "eks_cluster_subnets" {
+  description = "Subnets for EKS Cluster"
+  type        = list(string)
+  default     = ["subnet-xxxx", "subnet-yyyy"] // Provide a list of subnet-ids for Amazon EKS Cluster
+}
+
```
Comment (jackywong-amazon) [inline examples/resources/awscc_eks_cluster/eks_cluster_basic.tf:38]: Probably the best is to show your core TF configuration and remove those variables.
In the resource block, use example like
name = "example_cluster"

File: examples/resources/awscc_eks_cluster/eks_cluster_controlplane_logging.tf [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  description = "Subnets for EKS Cluster"
+  type        = list(string)
+  default     = ["subnet-xxxx", "subnet-yyyy"] // Provide a list of subnet-ids for Amazon EKS Cluster
+}
+
+variable "enabled_cluster_log_types" {
+  default     = ["api", "audit", "authenticator", "controllerManager", "scheduler"]
+  description = "A list of the Logs to be enabled in control plane."
+  type        = list(string)
+}
```
Comment (jackywong-amazon) [inline examples/resources/awscc_eks_cluster/eks_cluster_controlplane_logging.tf:43]: Same here,
Probably the best is to show your core TF configuration and remove those variables.
In the resource block, use example like
name = "example_cluster"
version = "1.1"

File: examples/resources/awscc_eks_cluster/eks_cluster_encrypt_kubernetes_secrets.tf [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+      Action = "sts:AssumeRole"
+      Effect = "Allow"
+      Principal = {
+        Service = "eks.amazonaws.com"
+      }
+    }]
+    Version = "2012-10-17"
+  })
+  managed_policy_arns = [
+    "${local.policy_arn_prefix}/AmazonEKSClusterPolicy",
```
Comment (jackywong-amazon) [inline examples/resources/awscc_eks_cluster/eks_cluster_encrypt_kubernetes_secrets.tf:56]: remove the local and show the exact policy arn

File: examples/resources/awscc_eks_cluster/eks_cluster_basic.tf (hunk 4 above)
Comment (maheshr-amzn) [reply examples/resources/awscc_eks_cluster/eks_cluster_basic.tf:38]: @jackywong-amazon Added in core TF configuration and removed the variables

File: examples/resources/awscc_eks_cluster/eks_cluster_controlplane_logging.tf (hunk 5 above)
Comment (maheshr-amzn) [reply examples/resources/awscc_eks_cluster/eks_cluster_controlplane_logging.tf:43]: @jackywong-amazon This has been taken care.

File: examples/resources/awscc_eks_cluster/eks_cluster_encrypt_kubernetes_secrets.tf (hunk 6 above)
Comment (maheshr-amzn) [reply examples/resources/awscc_eks_cluster/eks_cluster_encrypt_kubernetes_secrets.tf:56]: Removed and added exact policy arn

**Review by jackywong-amazon — APPROVED** [review]
LGTM, thanks for your contribution!!!

**Comment by ewbankkit** [comment]
@maheshr-amzn Thanks for the contribution :tada: :clap:.

**Pull request merged**